server/scsynth/SC_SequencedCommand.cpp

```diff
@@ -287,7 +288,9 @@ void BufAllocCmd::CallDestructor() { this->~BufAllocCmd(); }
 bool BufAllocCmd::Stage2() {
     SndBuf* buf = World_GetNRTBuf(mWorld, mBufIndex);
     mFreeData = buf->data;
-    SCErr err = bufAlloc(buf, mNumChannels, mNumFrames, mWorld->mSampleRate);
+    // use Server sample rate as default value
+    auto sampleRate = mSampleRate != 0.0 ? mSampleRate : mWorld->mSampleRate;
+    SCErr err = bufAlloc(buf, mNumChannels, mNumFrames, sampleRate);
```

Comment:
Make sure that `bufAlloc` returns `kSCErr_Failed` for negative sample rate values.